### PR TITLE
Added integration tests for snow sql

### PR DESCRIPTION
**Pull request by sfc-gh-astus**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [x] I've added new automated tests to verify correctness of my new code.
   * [x] I've confirmed that my changes are up-to-date with the target branch.
   * [x] I've described my changes in the section below.

### Changes description
Changed result print for snow sql command
Added integration tests for snow sql
Added utils to extract data from print_db_cursor mock

### CODE REVIEW

File: src/snowcli/cli/sql.py [hunk 1]
```diff
@@ -112,18 +79,9 @@ def execute_sql(
...
-        conn.ctx.execute_string(
-            sql_text=sql,
-            remove_comments=True,
-        )
+    results = conn.ctx.execute_string(
+        sql_text=sql,
+        remove_comments=True,
+    )
+    for result in results:
+        print_db_cursor(result)
```
Comment (sfc-gh-turbaszek) [inline src/snowcli/cli/sql.py:87]: Shouldn't it be `print_db_cursor(results)`?

File: tests_integration/test_files/sql_multi_queries.sql [hunk 2]
```diff
@@ -0,0 +1,2 @@
+select pi();
```
Comment (sfc-gh-turbaszek) [inline tests_integration/test_files/sql_multi_queries.sql:1]: Does this spin warehouse?

**Review by sfc-gh-turbaszek — APPROVED** [review]

File: src/snowcli/cli/sql.py (hunk 1 above)
Comment (sfc-gh-astus) [reply src/snowcli/cli/sql.py:87]: No, `print_db_cursor` expects `SnowflakeCursor`, but in this case it gets list of SnowflakeCursors

Comment (sfc-gh-turbaszek) [reply src/snowcli/cli/sql.py:87]: How it works with json format?

File: tests_integration/test_files/sql_multi_queries.sql (hunk 2 above)
Comment (sfc-gh-astus) [reply tests_integration/test_files/sql_multi_queries.sql]: changed to ln

File: src/snowcli/cli/sql.py (hunk 1 above)
Comment (sfc-gh-astus) [reply src/snowcli/cli/sql.py:87]: It treats each result separately 
```
% snow --format JSON sql -q "select 1; select pi(); select current_user"
[
  {
    "1": 1
  }
]
[
  {
    "PI()": 3.141592654
  }
]
[
  {
    "CURRENT_USER": "ADMIN"
  }
]
```

Comment (sfc-gh-turbaszek) [reply src/snowcli/cli/sql.py:87]: Can you pipe it to `jq`?

Comment (sfc-gh-astus) [reply src/snowcli/cli/sql.py:87]: no

**Pull request merged**